resource/resource_graph.go

```diff
@@ -641,6 +662,21 @@ func (g *Graph) SubGraphFrom(node Name) (*Graph, error) {
 	return subGraph, nil
 }
 
+// MarkReachability marks all nodes in the subgraph from the given [Name] node as either reachable [true] or unreachable [false].
+func (g *Graph) MarkReachability(node Name, reachable bool) error {
+	g.mu.Lock()
+	defer g.mu.Unlock()
+
+	subGraph, err := g.subGraphFrom(node)
+	if err != nil {
+		return err
+	}
+	for _, node := range subGraph.nodes {
+		node.unreachable = !reachable
```

Comment:
We don't need to take the node mutex because the graph is locked? I think that's risky.

Reply:
yeah, that's true - I'll add a private helper to set node reachability with a mutex

Reply:
done https://github.com/viamrobotics/rdk/pull/4421/commits/324ce54798b1d1e0d718e46cbf0113b6ea0c5480